includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -2276,4 +2338,161 @@ protected function guess_modal_role( DOMElement $modal ) {
 		// None of the roles we are looking for match any of the classes.
 		return Role::DIALOG;
 	}
+
+	/**
+	 * Evaluates the given XPath expression and give first Element if exist.
+	 *
+	 * @param string  $expression   The XPath expression to execute.
+	 * @param Element $context_node The optional context node can be specified for doing relative XPath queries.
+	 *                              By default, the queries are relative to the root element.
+	 *
+	 * @return Element|bool Return Element if exists otherwise False.
+	 */
+	protected function get_first_element( $expression, $context_node = null ) {
+
+		if ( empty( $expression ) ) {
+			return false;
+		}
+
+		$dom_node_list = $this->dom->xpath->query( $expression, $context_node );
+		if ( ! $dom_node_list instanceof DOMNodeList || 0 >= $dom_node_list->length ) {
+			return false;
+		}
+
+		/** @var Element $dom_node */
+		$dom_node = $dom_node_list->item( 0 );
+
+		return $dom_node;
+	}
+
+	/**
+	 * Update main navigation menu for "twentynineteen" theme.
+	 *
+	 * @return void
+	 */
+	public function update_twentynineteen_mobile_main_menu() {
+
+		if ( ! self::is_mobile_request() ) {
+			return;
+		}
+
+		$xpaths = [
+			'main_menu'               => '//nav[ @id = "site-navigation" ]//ul[ @class = "main-menu"]',
+			'top_menu_items'          => './li[ contains( @class, "menu-item" ) and contains( @class, "menu-item-has-children" ) ]',
+			'expand_button'           => './button[ @class = "submenu-expand" ]',
+			'submenu'                 => './ul[ @class = "sub-menu" ]',
+			'close_button_in_submenu' => './li[ contains( @class, "mobile-parent-nav-menu-item" ) ]//button[ contains( @class, "menu-item-link-return" ) ]',
+		];
+
+		$main_menu = $this->get_first_element( $xpaths['main_menu'] );
+		if ( empty( $main_menu ) ) {
+			return;
+		}
+
+		$menu_items = $this->dom->xpath->query( $xpaths['top_menu_items'], $main_menu );
+
+		/** @var Element $menu_item */
+		foreach ( $menu_items as $index => $menu_item ) {
+			$expand_button = $this->get_first_element( $xpaths['expand_button'], $menu_item );
+			$sub_menu      = $this->get_first_element( $xpaths['submenu'], $menu_item );
+
+			if ( empty( $expand_button ) || empty( $sub_menu ) ) {
+				continue;
+			}
+
+			$menu_item->setAttribute( 'class', $menu_item->getAttribute( 'class' ) . ' off-canvas' );
+
+			$sidebar_id = "primary_nav_$index";
+
+			// AMP Sidebar.
+			$amp_sidebar = AMP_DOM_Utils::create_node(
+				$this->dom,
+				Extension::SIDEBAR,
+				[
+					Attribute::ID     => $sidebar_id,
+					Attribute::LAYOUT => 'nodisplay',
+					Attribute::CLASS_ => 'amp-twentynineteen-main-navigation',
```

Comment:
To match the behavior of core, the sidebar should fly in from the right:
```suggestion
					Attribute::CLASS_ => 'amp-twentynineteen-main-navigation',
					Attribute::SIDE   => is_rtl() ? 'left' : 'right',
```